File: week11/Clock.py
from Counter import Counter
class Clock:
    def __init__(self):
        self._hour=Counter("Hour")
        self._min=Counter("Minute")
        self._sec=Counter("Second")

    def tick(self):
        if self._sec.ticks<59:
            self._sec.increment()
        else:
            self._sec.reset()
            if self._min.ticks<59:
                self._min.increment()
            else:
                self._min.reset()
                if self._hour.ticks<11:
                    self._hour.increment()
                else:
                    self._hour.reset()

    def reset(self):
        self._hour.reset()
        self._min.reset()
        self._sec.reset()

    @property
    def clock_time(self):
        return f"{self._hour.ticks:02}:{self._min.ticks:02}:{self._sec.ticks:02}"

File: week3/Task3_1P/Task3_1P/Clock.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

public class Clock
{
    private Counter _hour;
    private Counter _min;
    private Counter _sec;

    public Clock()
    {
        _hour = new Counter("Hour");
        _min = new Counter("Minute");
        _sec = new Counter("Second");
    }

    public void Tick()
    {
        if (_sec.Ticks <59)
        {
            _sec.Increment();
        }
        else
        {
            _sec.Reset();
            if(_min.Ticks <59)
            {
                _min.Increment();
            }
            else
            {
                _min.Reset();
                if(_hour.Ticks <11)
                {
                    _hour.Increment();
                }
                else
                {
                    _hour.Reset();
                }
            }
        }
    }

    public void Reset()
    {
        _hour.Reset();
        _min.Reset();
        _sec.Reset();
    }

    public string ClockTime
    {
        get
        {
            return $"{_hour.Ticks:D2}:{_min.Ticks:D2}:{_sec.Ticks:D2}";
        }
    }
}

